### PR TITLE
[OSPK8-421] Move OsMacAddr mgmt to osnetcfg

### PR DESCRIPTION
**Pull request by stuggi**
This moves the osmac reservations create to the osnetcfg controller
and updates the osmacaddr spec with new/changed reservations. For
this the osvmset and osbmset controller adds a label to its CR with
the osnetcfg CR name. This allows the osnetcfg to watch osbms and
osvmset CRs and reconciles on changes. It always recreates the full
list of reservations. Therefore no special handling is required
to cleanup reservations. If a node is gone from the osbms/osvmset
status also the mac reservation is gone.

The status of the osmacaddr CR still holds the same flattened
reservation list. This is still the source to render the ooo
parameter file.

This also changes the osbms to use the same common HostStatus type
to be able to use a common function to check the created osbms
and osvmset nodes. Right now required parameters from osbms got
merged into the existing HostStatus struct. In a follow up this
should be cleaned up (CtlplaneIP).

### CODE REVIEW

File: api/v1beta1/openstackmacaddress_types.go [hunk 1]
```diff
@@ -88,12 +98,14 @@ const (
 	MACCondReasonCreateMACError MACReason = "CreateMACError"
 	// MACCondReasonAllMACAddressesCreated - all MAC addresses created
 	MACCondReasonAllMACAddressesCreated MACReason = "MACAddressesCreated"
+	// MACCondReasonError - General error getting the OSMACaddr object
+	MACCondReasonError VMSetProvisioningReason = "MACError"
```
Comment (abays) [inline api/v1beta1/openstackmacaddress_types.go:102]: s/VMSetProvisioningReason/MACReason

File: controllers/openstackbaremetalset_controller.go [hunk 2]
```diff
@@ -1397,11 +1407,20 @@ func (r *OpenStackBaremetalSetReconciler) createNewHostnames(
 			return newHostnames, err
 		}
 
+		if instance.Status.BaremetalHosts == nil {
+			instance.Status.BaremetalHosts = map[string]ospdirectorv1beta1.HostStatus{}
+		}
```
Comment (abays) [inline controllers/openstackbaremetalset_controller.go:1412]: It looks like we're already doing this on lines 121-123.  Or maybe it is possible for the instance to arrive at the `createNewHostnames` func without hitting those lines above?

File: controllers/openstackvmset_controller.go [hunk 3]
```diff
@@ -1124,6 +1123,10 @@ func (r *OpenStackVMSetReconciler) createNewHostnames(
 			return newHostnames, err
 		}
 
+		if instance.Status.VMHosts == nil {
+			instance.Status.VMHosts = map[string]ospdirectorv1beta1.HostStatus{}
+		}
```
Comment (abays) [inline controllers/openstackvmset_controller.go:1128]: Same question here regarding this initialization as with BaremetalSet controller

File: controllers/openstackbaremetalset_controller.go (hunk 2 above)
Comment (stuggi) [reply controllers/openstackbaremetalset_controller.go:1412]: yeah, I have still seen a assignment to entry in nil map error. need to debug why, or just add them here for safety

File: controllers/openstackvmset_controller.go (hunk 3 above)
Comment (stuggi) [reply controllers/openstackvmset_controller.go:1128]: same

**Pull request merged**